src/flisp/Makefile

```diff
@@ -48,20 +58,22 @@ debug: $(BUILDDIR)/$(EXENAME)-debug
 $(BUILDDIR):
 	mkdir -p $(BUILDDIR)
 
-$(BUILDDIR)/%.o: %.c $(HEADERS) | $(BUILDDIR)
+$(BUILDDIR)/%.o: $(SRCDIR)/%.c $(HEADERS) | $(BUILDDIR)
 	@$(call PRINT_CC, $(CC) $(CPPFLAGS) $(SHIPFLAGS) $(DISABLE_ASSERTIONS) -c $< -o $@)
-$(BUILDDIR)/%.dbg.obj: %.c $(HEADERS) | $(BUILDDIR)
+$(BUILDDIR)/%.dbg.obj: $(SRCDIR)/%.c $(HEADERS) | $(BUILDDIR)
 	@$(call PRINT_CC, $(CC) $(CPPFLAGS) $(DEBUGFLAGS) -c $< -o $@)
 
-$(BUILDDIR)/flisp.o:   flisp.c cvalues.c types.c flisp.h print.c read.c equal.c
-$(BUILDDIR)/flisp.dbg.obj:  flisp.c cvalues.c types.c flisp.h print.c read.c equal.c
-$(BUILDDIR)/flmain.o:  flmain.c flisp.h
-$(BUILDDIR)/flmain.dbg.obj: flmain.c flisp.h
+FLISP_SRCS := $(flisp.c cvalues.c types.c flisp.h print.c read.c equal.c:%=$(SRCDIR)/%)
+FLMAIN_SRCS := $(flmain.c flisp.h:%=$(SRCDIR)/%)
+$(BUILDDIR)/flisp.o: $(FLISP_SRCS)
+$(BUILDDIR)/flisp.dbg.obj: $(FLISP_SRCS)
+$(BUILDDIR)/flmain.o: $(FLMAIN_SRCS)
+$(BUILDDIR)/flmain.dbg.obj: $(FLMAIN_SRCS)
 
-$(LLT_release): $(LLTDIR)/*.h $(LLTDIR)/*.c
-	$(MAKE) -C $(LLTDIR) BUILDDIR='$(abspath $(BUILDDIR)/$(LLTDIR))'
-$(LLT_debug): $(LLTDIR)/*.h $(LLTDIR)/*.c
-	$(MAKE) debug -C $(LLTDIR) BUILDDIR='$(abspath $(BUILDDIR)/$(LLTDIR))'
+$(LLT_release): $(LLTSRCDIR)/*.h $(LLTSRCDIR)/*.c
+	$(MAKE) -C $(NATIVE_BUILDDIR)/$(LLTDIR) $(subst $(abspath $(NATIVE_BUILDDIR)/$(LLTDIR))/,,$(abspath $(LLT_release)))
+$(LLT_debug): $(LLTSRCDIR)/*.h $(LLTSRCDIR)/*.c
+	$(MAKE) -C $(NATIVE_BUILDDIR)/$(LLTDIR) $(subst $(abspath $(NATIVE_BUILDDIR)/$(LLTDIR))/,,$(abspath $(LLT_debug)))
```

Comment:
Isn’t this slightly conflicted? The directory seems to be for native, but the subst looks for a slightly different prefix? Maybe we should add the abspath rule to the LLT Makefile, so we can remove the subst from here?

Reply:
I don't think there's much to be gained. Keeping the targets appropriately relative seems simplest to me. This just turns the absolute path back into a relative path.

Reply:
sure, just seemed like it wasn't relative to anything obviously meaningful

Reply:
Well, it's just "relative" to the LLT build dir, which seems reasonably meaningful.